### ISSUE TITLE
Compile error: Cannot find module '@angular/common/http/src/client' or HttpObserve in addBody function

### ISSUE THREAD

**Issue by Ali-ModernBeauty** [diegomvh/angular-odata#9]
Hey Diego! me again :D

File `node_modules/angular-odata/lib/client.d.ts`. Line `13`
```
export declare const addBody: <T>(options: {
    etag?: string;
    headers?: HttpHeaders | {
        [header: string]: string | string[];
    };
    observe?: import("@angular/common/http/src/client").HttpObserve; // <--- error here
    params?: HttpParams | {
        [param: string]: string | string[];
    };
    reportProgress?: boolean;
    responseType?: "text" | "blob" | "arraybuffer" | "json";
    withCredentials?: boolean;
}, body: T) => any;
```

![image](https://user-images.githubusercontent.com/38334720/78414643-61104d00-75da-11ea-8816-406c7bb3ce08.png)

Looks like `HttpObserve` is in `node_modules/@angular/common/http/http.d.ts` but isn't exported.

I believe this was a result of a recent patch. It was not there in `0.7.0` which is the version I upgraded from to `0.7.2`. (thank you for the other patches by the way).

For now, I solved compiling error _temporarily_ by editing the `node_modules/angular-odata/lib/client.d.ts`  by copying over the type values from Angular so it looks like this:
`observe?: 'body' | 'events' | 'response';`

_FYI and probably irrelative:_ Application was working fine with this error in Angular 8 before Ivy. It only loaded the second time it compiles when running `ng serve`, then saving a changed file. Now that I upgraded to Angular 9, I can't anymore.

So, do I need to install the Angular source code maybe? Because I see you're referencing `/src/client`

**Comment by diegomvh** [comment]
It is a strange error, since looking in the code there are no references to `HttpObserve` in the client.ts file.
I remember making some references to angular's client.ts but it was a while ago and these references are not present in the most current versions of the code ... I don't know what to think about it.
On the other hand ... the version with the tag 0.7.2 was created to make the code work in Angular 7 ... I published it and it was the last release .... I think it is best to use versions of 0.8.x later.
I hope to have a new version soon since many changes have been added.
